tests/acceptance/shared/cluster.go

```diff
@@ -0,0 +1,317 @@
+
+}
+
+// deleteWorkload deletes a workload and asserts that the workload is deleted.
+func deleteWorkload(workload, filename string) error {
+	fmt.Println("\nRemoving", workload)
+	cmd := "kubectl delete -f " + filename + " --kubeconfig=" + KubeConfigFile
+
+	_, err := RunCommandHost(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to run kubectl delete: %v", err)
+	}
+
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(5 * time.Second)
+
+	for {
+		select {
+		case <-timeout:
+			return errors.New("workload deletion timed out")
+		case <-tick:
+			isDeleted, err := IsWorkloadDeleted(workload)
+			if err != nil {
+				fmt.Printf("failed to check if workload is deleted: %v", err)
+			} else if isDeleted {
+				return nil
+			}
+		}
+	}
+}
+
+// IsWorkloadDeleted returns true if the workload is deleted.
+func IsWorkloadDeleted(workload string) (bool, error) {
+	res, err := RunCommandHost("kubectl get all -A --kubeconfig=" + KubeConfigFile)
+	if err != nil {
+		return false, err
+	}
+
+	return !strings.Contains(res, workload), nil
+}
+
+// KubectlCommand return results from various commands, it receives an "action" , source and args.
+// it already has KubeConfigFile
+//
+// destination = host or node
+//
+// action = get,describe...
+//
+// source = pods, node , exec, service ...
+//
+// args   = the rest of your command arguments.
+func KubectlCommand(destination, action, source string, args ...string) (string, error) {
+	var cmd string
+	var res string
+	var err error
+	kubeconfigFlag := " --kubeconfig=" + KubeConfigFile
+
+	if destination == "host" {
+		cmd = addKubectlCommand(action, source, args) + kubeconfigFlag
+		res, err = RunCommandHost(cmd)
+		if err != nil {
+			return "", err
+		}
+	} else if destination == "node" {
+		cmd = addKubectlCommand(action, source, args) + kubeconfigFlag
+		ips := FetchNodeExternalIP()
+		for _, ip := range ips {
+			res, err = RunCommandOnNode(cmd, ip)
+			if err != nil {
+				return "", err
+			}
+		}
+	} else {
+		return "", fmt.Errorf("invalid destination: %s", destination)
+	}
+
+	return res, nil
+}
+
+// addKubectlCommand using a specific action + source maps the args received  to create a kubectl command.
+func addKubectlCommand(action, source string, args []string) string {
+	shortCmd := map[string]string{
+		"get":      "kubectl get",
+		"describe": "kubectl describe",
+		"exec":     "kubectl exec",
+		"delete":   "kubectl delete",
+		"apply":    "kubectl apply",
+	}
+
+	cmdPrefix, ok := shortCmd[action]
+	if !ok {
+		cmdPrefix = action
+	}
+
+	return cmdPrefix + " " + source + " " + strings.Join(args, " ")
+}
+
+// Nodes returns the list of nodes in the cluster and parses the output with parseNodes.
+func Nodes(print bool) ([]Node, error) {
+	return parseNodes("kubectl get nodes --no-headers -o wide --kubeconfig="+KubeConfigFile, print)
+}
+
+// WorkerNodes returns the list of worker nodes in the cluster.
+func WorkerNodes(print bool) ([]Node, error) {
```

Comment:
Can we just use Roles instead of complex kubectl command here as it is already retrieved  in the structure?

Reply:
deleted this one. 

done